### DISCUSSION TITLE
What kind of Variable Message Signs should we support?

### DISCUSSION THREAD

**Discussion by emiltin** [rsmp-nordic/rsmp_sxl_vms#2] [Ideas]
RSMP Nordic is going to support Variable Message Signs (VMS'es) with a standardized SXL which we (RSMP Nordic) will maintain.

Different types of VMS exists. Which types should we support? Our aim is not to support everything, but rather to find a sweet spot where we support 80-90% of what road authorities need, without having to spend a lot of energy on supporting very specialized equipment.

- Simple fixed signs that can be turned on/off
- Text/ASCII based signs
- Full image signs, where you work with bitmaps
- HTML based?
- Signs with multiple sections/signs

**Comment by emiltin** [comment]
Do we need to support sign that consists of multiple sections? That would require some sort of concept of sections within a single sign, and a concept for how writing to such a group of sections would work. Or should keep it simple and define a VMS as a single sign we no concept of sub sections.

**Comment by otterdahl** [comment]
The example which were [posted a couple of years](https://github.com/rsmp-nordic/rsmp_sxl_variable_message_sign/blob/first_draft/SXL_Variable_Message_Sign_ver_0_1-2021-07-01.yaml) ago is a very common SXL for VMS signs in Sweden. It only consists of a single, bitmapped section.

The thing which might be missing however, is alarm for UPS. UPS is also quite common.

**Comment by yco-svms** [comment]
_Let me introduce myself. I've been working for SVMS in France as Head of Innovation for the last 2 years. Before that, I was with the company for 18 years, where I was involved in developing traffic station, weather and VMS products.
In my current role, I'm working on the adoption of a new communication protocol for our products: RSMP._

_In the rest of this post, I'll give you the current state of my thinking on the subject._

Freezing the type of VMS may result in the need to define as many SXLs as there are types of VMS.

From my point of view, I would prefer a generic design for an SXL for VMS.

With this in mind, I propose :
- to separate the VMS into: what it does and how it is built to do it
- define what a VMS does: display or not display at least one message according to an animation scenario
- to separate display control from function control (status, faults, etc.)

At the end of this work, we can define a generic SXL and a cross-tabulation between VMS types and mandatory, optional or unsuitable capabilities.

### Initial assumptions _(in bulk)_
- A VMS is a device that dynamically displays a message by activating points distributed around its display area.
- These points:
   - are distributed in a matrix (or not?)
   - can be displayed in monochrome, multichrome (a few predefined colours) or full colour (RGB)
- The display area can be divided into several statically or dynamically defined sub-areas (it contains at least one sub-area that corresponds to the display area)
- A message (or group of messages) is displayed in a sub-zone
- The messages can represent images (pictograms, photos, etc.) or text in different alphabets (and orientation)
- Animations on groups of at least one message can be available: fixed, flashing (alternating with an "empty" message), alternating between several messages, scrolling (?)
- VMS follow display rules that depend on the regions (authorities) where they are in operation:
  - type of animations allowed
  - message composition rules to ensure optimum readability depending on the VMS role (the role defines the message readability constraints depending on the reading distance and the approach speed of the message recipient)
 
_... to be continued..._

**Reply by emiltin** [reply]
thanks for the input!

**Reply by emiltin** [reply]
i've tried to create a recap at #4
